training/train_yolo.py

```
@@ -1,72 +1,161 @@
-    root_dir = Path(__file__).resolve().parent.parent
-    print(root_dir)
-    config_path = (root_dir / "training/training_config.yaml").resolve()
-    config = load_config(config_path)
+        # Resolve path to dataset YAML
+        dataset_config = config_dict.get("data")
+        if dataset_config is None:
+            self.dataset_path = (ROOT_DIR / "training" / "data" / "data.yaml").resolve()
+        else:
+            self.dataset_path = (ROOT_DIR / config["data"]).resolve()
+        if not self.dataset_path.exists():
+            raise FileNotFoundError(f"Dataset configuration file not found: {self.dataset_path}")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)